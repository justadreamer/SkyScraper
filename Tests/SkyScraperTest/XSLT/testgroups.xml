<?xml version="1.0"?>
<a>
    <c>a12b</c>
    <c>c23d</c>
    <emoji>ssðŸŒ±ee</emoji>
</a>
