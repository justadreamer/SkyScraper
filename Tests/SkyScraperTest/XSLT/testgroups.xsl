<?xml version="1.0"?>
<xsl:stylesheet version="1.0"
xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
xmlns:regexp="http://exslt.org/regular-expressions"
extension-element-prefixes="regexp">
<xsl:output method="text" />
<xsl:template match="c">
    <xsl:variable name="results" select="regexp:match(string(.), '[^\d]*(\d*)[^\d]*', 'g')" />
    <xsl:value-of select="$results[1]" /><xsl:value-of select="$results[2]" /><xsl:value-of select="$results[2]" />
</xsl:template>
<xsl:template match="emoji">
    <xsl:value-of select="regexp:replace(string(.),'s+([^e]*)e+','g','$1$1')" />
</xsl:template>
</xsl:stylesheet>