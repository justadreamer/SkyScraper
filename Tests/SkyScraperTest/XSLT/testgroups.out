
    a12b1212
    c23d2323

